nkms_eth/miner.py

```diff
@@ -0,0 +1,85 @@
+from typing import Tuple
+
+from .blockchain import Blockchain
+from .escrow import Escrow
+from .token import NuCypherKMSToken
+
+
+class Miner:
+    """Practically carrying a pickaxe"""
+
+    def __init__(self, blockchain: Blockchain, token: NuCypherKMSToken, escrow: Escrow, address=None):
+        self.blockchain = blockchain
+        self.token = token
+        self.address = address
+
+        self.escrow = escrow
+        if not escrow.contract:
+            raise Escrow.ContractDeploymentError('Escrow contract not deployed. Arm then deploy.')
+        else:
+            escrow.miners.append(self)
+
+    def __repr__(self):
+        class_name = self.__class__.__name__
+        r = "{}(address='{}')"
+        r.format(class_name, self.address)
+
+    def __del__(self):
+        self.escrow.miners.remove(self)
```

Comment:
This needs an explanation.

Reply:
At the moment - I am keeping a list of all miners on the escrow objects as a way to track them; however, I'm not certain this is the best way to achieve this. If you need to get every miner address, you can do so with `.swarm`, which uses cumulative sum via solidity to get each miner address.  thoughts?

 :heavy_check_mark: